compiler/rustc_mir/src/transform/simplify.rs

```diff
@@ -451,7 +459,23 @@ impl Visitor<'_> for UsedLocals {
         }
     }
 
-    fn visit_local(&mut self, local: &Local, _ctx: PlaceContext, _location: Location) {
+    fn visit_local(&mut self, local: &Local, ctx: PlaceContext, _location: Location) {
+        debug!("local: {:?} is_static: {:?}, ctx: {:?}", local, self.is_static, ctx);
+        // Do not count a local as used in `_local = <rhs>` if RHS is a ZST.
+        let store = matches!(ctx, PlaceContext::MutatingUse(MutatingUseContext::Store));
```

Comment:
The essence of the current algorithm is that it skips visitation of a `_local` in `_local = <rhs>` statements. This condition should always be false, except for an inline assembly where visitation process is not customized.

Reply:
Yep, tests still pass locally with this removed